src/lib.rs

```diff
@@ -158,6 +162,10 @@ impl Pty {
         // and it's not safe to keep it open
         libc::close(raw_controller_fd);
 
+        // just to be safe, mark every single file descriptor as close-on-exec.
+        #[cfg(target_os = "linux")]
+        libc::close_range(3, libc::c_uint::MAX, libc::CLOSE_RANGE_CLOEXEC as c_int);
```

Comment:
this is a linux-only syscall, and should let us close an fd that was accidentally leaked _all_ the way from pid1!!!!.

good news, the test still passes on macos.